File: src/main/java/com/hedvig/claims/events/DataItemAddedEvent.java
package com.hedvig.claims.events;

import com.hedvig.claims.web.dto.ClaimDataType.DataType;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
public class DataItemAddedEvent {

  private String id;
  private String claimsId;
  public LocalDateTime date;
  public String userId;

  public DataType type;
  public String name;
  public String title;
  public Boolean received;
  public String value;
}

File: src/main/java/com/hedvig/claims/events/DataItemAddedEvent.kt
package com.hedvig.claims.events

import com.hedvig.claims.web.dto.ClaimDataType.DataType
import lombok.Data

import java.time.LocalDateTime

data class DataItemAddedEvent(
  var id: String,

  var claimsId: String,
  var date: LocalDateTime,
  var userId: String,

  var type: DataType,
  var name: String,
  var title: String,
  var received: Boolean?,
  var value: String
)
